### ISSUE TITLE
link-to complex query param blows up

### ISSUE THREAD

**Issue by cowboyd** [emberjs/ember.js#5429]
If you have a query param that is mapped to a nested property, trying to include it in an invocation of `link-to` throws an exception. E.g.

``` js
App.IndexController = Ember.Controller.extend({
  queryParams: [{
    "sort.house": "sort"
  }]
})
```

``` handlebars
{{#link-to "index" (query-params sort="hufflepuff")}}Hufflepuff{{/link-to}}
```

results in the following error:

```
Uncaught Error: Assertion Failed: You're not allowed to have more than one controller property map to the same query param key, but both `index:sort.house` and `index:sort.house` map to `sort`. You can fix this by mapping one of the controller prop...
```

http://jsbin.com/kewam/3/edit

**Comment by bargar** [comment]
Inspired by some of the comments here, I found that a workaround for:

```
  queryParams: {
    controllerProperty: queryKey
  }
```

is:

```
  queryParams: [
    'queryKey'
  ],
  queryKey: Ember.computed.alias('controllerProperty')
```

Of course, this has the disadvantage of requiring you to have an explicit property called `queryKey` in your controller, but it's better than having to eschew having mapping when using `{{link-to}}` altogether.

**Comment by H1D** [comment]
@machty bug is still here (Ember 1.12)
click "post" to get wierd error
http://jsbin.com/sucew/19/edit?html,js,console,output
same version without QP-mapping (it works)
http://jsbin.com/sucew/20/edit?html,js,console,output

**Comment by H1D** [comment]
for future people from google search:
it won't be fixed due to upcoming refactoring
https://twitter.com/asuschev/status/601139210301644800

**Comment by myjpa** [comment]
Thanks @H1D  for your last comment, being hit the same wall and found this post from google search

**Comment by adambuczynski** [comment]
Likewise. Very problematic, creating a query param property in a base controller, causes a transitioning to a route with a child controller to fail with the above error.

**Comment by rwjblue** [comment]
This is fixed in Ember 2.8: http://output.jsbin.com/vudohuh
